storage/watchable_store.go

```diff
@@ -270,7 +272,7 @@ func (s *watchableStore) syncWatchers() {
 	for w, eb := range newWatcherBatch(&s.unsynced, evs) {
 		select {
 		// s.store.Rev also uses Lock, so just return directly
-		case w.ch <- WatchResponse{WatchID: w.id, Events: eb.evs, Revision: s.store.currentRev.main}:
+		case w.ch <- WatchResponse{WatchID: w.id, Events: eb.evs, Revision: s.store.currentRev.main, Progress: Progress{StartRevision: w.cur}}:
```

Comment:
My point about the `CompactRevision` stuff was that there's no need for a separate `Progress` field... the progress is a "virtual" `CompactRevision`